core/dbt/deps/registry.py

```diff
@@ -60,32 +57,7 @@ def _fetch_metadata(self, project, renderer) -> RegistryPackageMetadata:
         return RegistryPackageMetadata.from_dict(dct)
 
     def install(self, project, renderer):
-        metadata = self.fetch_metadata(project, renderer)
```

Comment:
moved upstream to PinnedPackage class so they can be shared across TarballPinnedPackage and RegistryPinnedPackage via PinnedPackage inheritance.